tmail-backend/mailets/src/main/scala/com/linagora/tmail/mailets/ContactsCollection.scala

```diff
@@ -55,9 +55,12 @@ class ContactsCollection @Inject()(eventBus: EventBus) extends GenericMailet {
 
   override def service(mail: Mail): Unit =
     if (CollectionUtils.isNotEmpty(mail.getRecipients)) {
-      val contacts: Seq[ContactFields] = extractRecipientsContacts(mail)
-      dispatchEvents(contacts)
-      appendAttributeToMail(mail, contacts)
+     SMono.justOrEmpty(mail.getMaybeSender.asOptional().toScala)
+        .map(sender => extractRecipientsContacts(mail) -> sender)
+        .flatMap {
+          case (contacts, sender) => SFlux.zip(appendAttributeToMail(sender, mail, contacts), dispatchEvents(sender, contacts))
```

Comment:
Should we check if the Sender is a user of James before dispatching the Event to EventBus?

I think we would not want to index stuff for outside sender like `elonmusk@tesla.com`.

Reply:
No not really generally this is coupled with SenderIs matcher